frontend/src/components/new-car-card/new-car-card.tsx

```diff
@@ -1,35 +1,100 @@
-import { FC } from 'react';
+import React from 'react';
 
-import bmw from '@assets/images/bmw_logo.svg';
 import compare from '@assets/images/compare.svg';
-import heart from '@assets/images/heart.svg';
-import new_car_1 from '@assets/images/mock_car_picture.png';
+import { WishlistInput } from '@autoline/shared/common/types/types';
+import { ExtendedCarCardPropsType } from '@common/types/types';
+import { Heart } from '@components/common/icons/icons';
+import { clsx } from 'clsx';
 
 import styles from './styles.module.scss';
 
-const NewCarCard: FC = () => {
+const NewCarCard: React.FC<ExtendedCarCardPropsType> = (props) => {
+  const formatPriceNumber = (num: number): string => {
+    const price = num.toString().split('.');
+    let numberPart = price[0];
+    const floatPart = price.length > 1 ? '.' + price[1] : '';
+    const rgx = /(\d+)(\d{3})/;
+    while (rgx.test(numberPart)) {
+      numberPart = numberPart.replace(rgx, '$1' + ' ' + '$2');
+    }
+    return numberPart + floatPart;
+  };
```

Comment:
This needs to be outside of the React component, in helpers/utils. Because this function definitely can be reused. Also, try to minimize the creation of the functions inside the react components because they are getting recreated on each rerender